src/Peachpie.Library.Graphics/PhpGd2.cs

```diff
@@ -667,16 +667,50 @@ public static long imagecolorallocatealpha(PhpResource im, int red, int green, i
         /// <summary>
         /// RGBA values to PHP Color format.
         /// </summary>
-        static long RGBA(long red, long green, long blue, long alpha = 0xff)
+        static long RGBA(long red, long green, long blue, long alpha = 0)
         {
             return (alpha << 24)
-                | ((blue & 0x0000FF) << 16)
+                | ((red & 0x0000FF) << 16)
                 | ((green & 0x0000FF) << 8)
-                | (red & 0x0000FF);
+                | (blue & 0x0000FF); 
         }
 
-        static Rgba32 FromRGB(long color) => new Rgba32((uint)color | 0xff000000u);
-        static Rgba32 FromRGBA(long color) => (color != (long)ColorValues.TRANSPARENT) ? new Rgba32((uint)color) : (Rgba32)Color.Transparent;
+        static Rgba32 FromRGB(long color) => new Rgba32((uint)color & 0x00ffffffu);
+        static Rgba32 FromRGBA(long color) => (color != (long)ColorValues.TRANSPARENT) ? new Rgba32((uint)ConvertAlpha(color, true)) : (Rgba32)Color.Transparent;
+
+        /// <summary>
+        /// Php Alpha is reversed map. PHP: 0 -> PNG,ImageSharp:255, PHP: 127 -> PNG,ImageSharp: 0. This method converts an alpha part of a color from PHP/ImageSharp to ImageSharp/PHP.
+        /// </summary>
+        /// <param name="FromPhp">True for PHP to ImageSharp else ImageSharp to PHP </param>
+        /// <returns>Converted color</returns>
+        static long ConvertAlpha(long color, bool FromPhp)
```

Comment:
Maybe make this as two methods as an extension for `Rgba32`. Avoid using `long` for everything.

```c#
Rgba32.ToLong(bool phpalpha)
```

Reply:
fixed it in commit Review fix. But I did only one extension because i think the extension for long is not good. ( ``` long.ToRgba32()```). So I did it as a static method (``` static Rgba32 PHPColorToRgba(uint color)```). Do you agree ? If not, I will repair it...

Reply:
right, extension only makes sense for `Rgba32`